### PR TITLE
Adding support for array split

### PR DESCRIPTION
**Pull request by AdrienCorenflos**
Adapted the split code to reproduce array_split behaviour

### CODE REVIEW

File: jax/numpy/lax_numpy.py [hunk 1]
```diff
@@ -1374,6 +1374,29 @@ def split(ary, indices_or_sections, axis=0):
...
+                                                 "in jax.numpy.array_split argument 1")
+    if indices_or_sections <= 0:
+      raise ValueError('number of sections must be larger than 0.')
+    part_size, r = _divmod(size, indices_or_sections)
+    split_indices = np.concatenate([[0], r * [part_size + 1], (indices_or_sections - r) * [part_size]])
+
+  starts, ends = [0] * ndim(ary), shape(ary)
+  _subval = lambda x, i, v: subvals(x, [(i, v)])
+  return [lax.slice(ary, _subval(starts, axis, start), _subval(ends, axis, end))
+          for start, end in zip(split_indices[:-1], split_indices[1:])]
```
Comment (jakevdp) [inline jax/numpy/lax_numpy.py:1397]: A lot of this function is identical to `split`. My understanding is that `array_split` only differs from `split` for integer `indices_or_sections` that are not divisors of the corresponding axis. Could we compute `split_indices` in that case and delegate the rest of this function to `split()`?

File: tests/lax_numpy_test.py [hunk 2]
```diff
@@ -1937,15 +1937,18 @@ def testFullLike(self, shape, in_dtype, fill_value_dtype, out_dtype, rng_factory
...
+      )
+  def testSplitStaticInt(self, shape, num_sections, axis, dtype, rng_factory, split_fun_name):
     rng = rng_factory(self.rng())
-    np_fun = lambda x: np.split(x, num_sections, axis=axis)
-    jnp_fun = lambda x: jnp.split(x, num_sections, axis=axis)
+    np_fun = lambda x: getattr(np, split_fun_name)(x, num_sections, axis=axis)
+    jnp_fun = lambda x: getattr(jnp, split_fun_name)(x, num_sections, axis=axis)
     args_maker = lambda: [rng(shape, dtype)]
     self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker)
     self._CompileAndCheck(jnp_fun, args_maker)
```
Comment (jakevdp) [inline tests/lax_numpy_test.py:1954]: It looks like this test doesn't exercise the situation where the array size is not divisible by the number of sections.

**Review by jakevdp — CHANGES_REQUESTED** [review]
Looks good, thanks for the contribution! A couple comments below:

**Comment by jakevdp** [comment]
Done in #4197
